internal/reconciler/shoot/reconciler.go

```diff
@@ -23,10 +40,125 @@ type Reconciler struct {
 }
 
 // Reconcile handles reconciliation requests for Shoots marked to be trusted in the Garden cluster.
-func (r *Reconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result, error) {
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := logf.FromContext(ctx)
 
-	log.Info("Shoot reconcile finished")
+	ctx, cancel := controllerutils.GetMainReconciliationContext(ctx, controllerutils.DefaultReconciliationTimeout)
+	defer cancel()
+
+	shoot := &gardencorev1beta1.Shoot{}
+	if err := r.Client.Get(ctx, req.NamespacedName, shoot); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(1).Info("Object is gone, stop reconciling")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("error retrieving shoot from store: %w", err)
+	}
+
+	if shoot.Annotations[v1beta1constants.AnnotationAuthenticationIssuer] != v1beta1constants.AnnotationAuthenticationIssuerManaged {
+		log.V(1).Info("Shoot does not have expected annotation or their value is not 'managed'", "annotation", v1beta1constants.AnnotationAuthenticationIssuer, "value", shoot.Annotations[v1beta1constants.AnnotationAuthenticationIssuer])
+		return reconcile.Result{}, nil
+	}
+
+	if shoot.Annotations[AnnotationTrustedShoot] != "true" && shoot.Annotations[AnnotationTrustedShoot] != "false" {
+		log.V(1).Info("Shoot does not have expected annotation or their value is not 'true'/'false'", "annotation", AnnotationTrustedShoot, "value", shoot.Annotations[AnnotationTrustedShoot])
+		return reconcile.Result{}, nil
+	}
+
+	if shoot.Annotations[AnnotationTrustedShoot] == "false" {
+		log.V(1).Info("Shoot has trusted annotation set to 'false', clean up OIDC resource")
+		return r.deleteOIDC(ctx, log, shoot)
+	}
```

Comment:
WDYT of calling `deleteOIDC` when the annotation != "true". Right now 

1. I can set it to "true" and we will create the OIDC resource
2. I set it to "foo", but we will not remove it.


We can plan of implementing a garbage collection controller later on, but it would be better IMO if such resources are deleted immediately.